
This patch temporarily reverts:

----
From 5e5aeb4367b450a28f447f6d5ab57d8f2ab16a5f Mon Sep 17 00:00:00 2001
From: Sasha Levin <sasha.levin@oracle.com>
Date: Wed, 3 Dec 2014 19:25:05 -0500
Subject: time: adjtimex: Validate the ADJ_FREQUENCY values
----

as it breaks userspace, atleast chrony.
https://bugs.mageia.org/show_bug.cgi?id=15174

Upstream is tracking it and it will be fixed, either in
kernel or in chrony, but for now revert it.

Signed-off-by: Thomas Backlund <tmb@mageia.org>

---
 kernel/time/ntp.c |    7 +++++++
 1 file changed, 7 insertions(+)

--- linux/kernel/time/ntp.c.orig
+++ linux/kernel/time/ntp.c
@@ -633,13 +633,6 @@ int ntp_validate_timex(struct timex *txc
 	if ((txc->modes & ADJ_SETOFFSET) && (!capable(CAP_SYS_TIME)))
 		return -EPERM;
 
-	if (txc->modes & ADJ_FREQUENCY) {
-		if (LONG_MIN / PPM_SCALE > txc->freq)
-			return -EINVAL;
-		if (LONG_MAX / PPM_SCALE < txc->freq)
-			return -EINVAL;
-	}
-
 	return 0;
 }
 
